Framework/API/src/Algorithm.cpp

```diff
@@ -116,7 +116,7 @@ Algorithm::Algorithm()
       m_communicator(std::make_unique<Parallel::Communicator>()), m_properties() {}
 
 /// Virtual destructor
-Algorithm::~Algorithm() {}
+Algorithm::~Algorithm() = default;
```

Comment:
Just out of interest. Is there any reason we need to define the constructor at all and not let the compiler generate it as it's `default`?

Reply:
In this case it's clang-tidy removing the stub implementation and replacing it with an explicit default

I can do a follow up PR to remove all instances of `T::~T() = default` (which don't override) in the codebase?

Reply:
Ah yes I see.

I think it might be worth it or someone else will likely come along with the same question as to why it's defaulted. This should have the bonus of inlining it?

Reply:
Do we know why this is marked override but it doesn't do anything? The base class `IPropertyManager.h` has a virtual destructor so this shouldn't need one?

Reply:
We've got quite a few with trivial overridden destructors that I can see in the changeset I think?

Reply:
Originally I tried removing them, but it resulted in several hundred linker errors. 
So ultimately it was easier for me to replace `~T() override {} = default;` and rely on future developers to notice that the destructor in both places are trivial and can be removed as/when they touch the code